tls.c

```diff
@@ -107,11 +107,19 @@ static bool load_server_certificates(char **errmsg) {
     } else if (!SSL_CTX_check_private_key(settings.ssl_ctx)) {
         snprintf(error_msg, errmax, "Error validating the certificate\r\n");
         success = false;
-    } else {
-        settings.ssl_last_cert_refresh_time = current_time;
+    } else if (settings.ssl_ca_cert) {
+        SSL_CTX_set_client_CA_list(settings.ssl_ctx,
+          SSL_load_client_CA_file(settings.ssl_ca_cert));
+        if (!SSL_CTX_load_verify_locations(settings.ssl_ctx,
+          settings.ssl_ca_cert, NULL)) {
+            snprintf(error_msg, errmax,
+              "Error loading the CA certificate: %s\r\n", settings.ssl_ca_cert);
+            success = false;
+        }
     }
     SSL_UNLOCK();
     if (success) {
+        settings.ssl_last_cert_refresh_time = current_time;
```

Comment:
While you are at this, you may remove the line https://github.com/memcached/memcached/blob/master/tls.c#L169. Timestamp update is repetitive there.